variables.tf

```diff
@@ -0,0 +1,86 @@
+
+variable "service_principal_name" {}
+variable "service_principal_secret" {}
+
+variable "agent_count" {
+  default = 3
+}
+
+variable "cluster_name" {
+  type        = string
+  default     = null
+  description = "The Domino cluster name for the K8s cluster and resource group"
+}
+
+variable "ssh_public_key" {
+  default = "~/.ssh/id_rsa.pub"
+}
+
+variable "dns_prefix" {
+  default = "k8stest"
+}
+
+variable "location" {
+  default = "West US 2"
+}
+
+variable "log_analytics_workspace_name" {
+  default = "testLogAnalyticsWorkspaceName"
+}
+
+# refer https://azure.microsoft.com/global-infrastructure/services/?products=monitor for log analytics available regions
+variable "log_analytics_workspace_location" {
+  default = "eastus"
+}
+
+# refer https://azure.microsoft.com/pricing/details/monitor/ for log analytics pricing 
+variable "log_analytics_workspace_sku" {
+  default = "PerGB2018"
+}
+
+variable "node_pools" {
+  type = map(object({
+    vm_size                        = string
+    zones                          = list(string)
+    node_os                        = string
+    taints                         = list(string)
+    cluster_auto_scaling           = bool
+    cluster_auto_scaling_min_count = number
+    cluster_auto_scaling_max_count = number
+  }))
+  default = {
+    compute = {
+      vm_size                        = "Standard_DS2_v2"
+      zones                          = ["1", "2", "3"]
+      node_os                        = "Linux"
+      taints                         = null
+      cluster_auto_scaling           = true
+      cluster_auto_scaling_min_count = 1
+      cluster_auto_scaling_max_count = 4
+    }
+    gpu = {
+      vm_size                        = "Standard_DS2_v2"
+      zones                          = ["1", "2", "3"]
+      node_os                        = "Linux"
+      taints                         = null
+      cluster_auto_scaling           = true
+      cluster_auto_scaling_min_count = 1
+      cluster_auto_scaling_max_count = 1
+    }
+    platform = {
+      vm_size                        = "Standard_DS2_v2"
+      zones                          = ["1", "2", "3"]
+      node_os                        = "Linux"
+      taints                         = null
+      cluster_auto_scaling           = true
+      cluster_auto_scaling_min_count = 1
+      cluster_auto_scaling_max_count = 4
+    }
+  }
+}
+
+variable subscription_id {
+  type        = string
+  default     = "de806df8-4359-4802-b814-b8a7699cfaa5" # Domino Engineering Platform Dev
```

Comment:
Subscription ID's don't need to be private. Is this coming from a security concern? I just want to make it as easy as possible to run the tf module directly/manually (it makes debugging easier even if the preferred path is through the deployer).

Reply:
Yeah, this is a public repo and at the end of the day, all Domino usage of it is going to be via the deployer (and probably via fleetcommand) and any usage outside of that context would likely be because we gave it to a customer as an example, so I'd rather not have the id committed unnecessarily to the public repo when it can just be in the deployer.

Reply:
Actually, I believe if you don't specify it, it's able to infer it from your account anyway. Definitely works this way when working from a service principal (and it's how we've been doing the azure rancher version).

Reply:
Fwiw, I do think it'd be better to have the module be as Domino-less as possible. On a similar note, maybe even the [VPN IPs](https://github.com/cerebrotech/terraform-gcp-gke/blob/44ade6c588f5a8e38402d16436afe27cb7538dc4/variables.tf#L143-L163) and [project ID](https://github.com/cerebrotech/terraform-gcp-gke/blob/44ade6c588f5a8e38402d16436afe27cb7538dc4/variables.tf#L15) don't necessarily belong in a public repo / module?